pages/api/auth/[...nextauth].js

```diff
@@ -42,4 +56,42 @@ const authOptions = {
   }
 }
 
+/**
+ * Takes a token, and returns a new token with an updated `accessToken`.
+ * If an error occurs, returns the old token and an error property
+ */
+const refreshAccessToken = async (token) => {
+  try {
+    const url = `https://${process.env.NEXT_PUBLIC_PROVIDER_NAME}.scientist.com/oauth/token`
+    const encodedString = Buffer.from(`${process.env.CLIENT_ID}:${process.env.CLIENT_SECRET}`).toString('base64')
+    const params = new URLSearchParams({
+      grant_type: 'refresh_token',
+      refresh_token: token.refreshToken,
+    })
+
+    const response = await axios.post(url, params, {
+      headers: {
+        'Authorization': `Basic ${encodedString}`,
+      },
+    })
+    const refreshedTokens = response.data
+
+    if (response.status !== 200) {
```

Comment:
are there times where this could be 201? or always 200?

Reply:
as far as I know...it'll always be 200.